src/main/scala/io/netty/handler/ssl/util/SelfSignedCert.scala

```diff
@@ -0,0 +1,30 @@
+package io.netty.handler.ssl.util
+
+import java.io.File
+import java.security.{KeyPairGenerator, SecureRandom}
+
+
+/**
+ * A container for a self-signed certificate and its private key.
+ *
+ * We use netty's code for generating self-signed certificates, but specialized to always
+ * use their bouncy castle generator, which uses SHA256 to sign the cert, instead of SHA1
+ * as used by their jdk certificate generator.
+ *
+ * We put this in the io.netty.handler.ssl.util package so that we can access the package
+ * protected BouncyCastleSelfSignedCertGenerator class.
+ */
+case class SelfSignedCert(certificate: File, privateKey: File)
```

Comment:
You're right that it's not really needed here. I tend to default to case classes unless the 1) class is mutable or 2) the class does not have value semantics. The cert and key files are read only, so they are as immutable as we can hope files to be, and two self-signed cert instances with the same cert and key files should be equal, so it has value semantics. Hence in my view, a case class is appropriate here, even if we don't actually use a lot of the syntactic niceties like pattern matching that case classes enable. My bias in scala is to default to immutability when possible, and that leads me to default to case classes as well.